cypress/e2e/gallery.cy.ts

```diff
@@ -0,0 +1,42 @@
+describe('Gallery page', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000/gallery')
+  })
+
+  it('should view image on thumbnail click', () => {
+    cy.get('[data-test=gallery-thumbnail-wrap]')
+      .find('[data-test=carousel-item-4]')
+      .as('fourthIndexThumbnail')
+
+    cy.get('[data-test=gallery-viewer-wrap]')
+    .find('[data-test=carousel-item-4]')
+    .as('fourthIndexImage')
+
+    cy.get('@fourthIndexImage')
+      .should('not.be.visible')
+
+    cy.get('@fourthIndexThumbnail')
+      .click()
+
+    cy.get('@fourthIndexImage')
+      .should('be.visible')
+  })
+
+  it('should view images on navigation button clicks', () => {
+    cy.get('[data-test=nav-next]')
+      .click().click().click().click().click()
```

Comment:
Nit: for the `.click()` x 5 happening twice, is it worthwhile to wrap in a function?